tests/test_common.py

```diff
@@ -92,3 +48,63 @@ def test_stringify(args, expected):
 )
 def test_prepare_params(args, expected):
     assert prepare_params(args) == expected
+
+
+@pytest.mark.parametrize(
+    "network, expected",
+    [
+        ("localhost", "http://127.0.0.1:5050/"),
+        ("goerli", "https://alpha4.starknet.io/gateway"),
+        ("mainnet", "https://alpha-mainnet.starknet.io/gateway"),
+    ],
+)
+def test_get_gateway_url(network, expected):
+    if network == "localhost":
+        with patch("nile.common.GATEWAYS") as mock_gateways:
```

Comment:
not sure i understand why `localhost` is different

Reply:
When the network is `localhost`, a dictionary value is returned from the `get` method (`GATEWAYS.get(network)`). The alpha networks don't use a dict to get their url, so we just mock it for `localhost`...we should also add `goerli2` and `integration` in the test